### PR TITLE
stop CommandLineParser from recognizing abbreviations

### LINKED ISSUE #1347: Disable argument abbreviation in joptsimple
Argument abbreviation is causing confusing issues like https://github.com/broadinstitute/gatk/issues/1306, where `-V` means different things in different tools. Let's disable it in `joptsimple` (or, if there's no mechanism to disable it, ask the author to add one).

### PR DESCRIPTION
**Pull request by lbergelson**
previously CommandLineParser would accept non-ambiguous abbreviations for longopts
this caused issues where -v would have different meanings in different tools

I patched joptSimple to add an option to disallow abbreviations, and updated to 5.0-beta-1 which incorporates the necessary changes.
fixes #1347

### CODE REVIEW

**Comment by lbergelson** [comment]
hmn, looks like tests are failing because some of them were using abbreviations, will fix and then reassign when they're fixed.

**Comment by lbergelson** [comment]
@akiezun fixed the tests

File: src/test/java/org/broadinstitute/hellbender/tools/picard/sam/FilterReadsIntegrationTest.java [hunk 1]
```diff
@@ -112,8 +112,7 @@ public void testReadFilter(
         }
 
         if (!writeReadsFile) { // defaults to true
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/sam/FilterReadsIntegrationTest.java]: Something bogus here - the condition says `if not writeReadsFile then use --WRITE_READS_FILES`

**Comment by akiezun** [comment]
looks fine, but 2 more things before merge:
- fix the bogus things about `writeReadsFile` in `FilterReadsIntegrationTest.java`
- add a test that shows that we do not accept abbreviations, ie that the PR does what it claim "stop CommandLineParser from recognizing abbreviations"

**Pull request merged**